src/services/csv.js

```diff
@@ -1,29 +1,36 @@
 const { ipcMain } = require('electron');
 const fs = require('fs');
 import { parse } from 'csv-parse';
+import { CSV_PARSE_ERROR } from '../components/InvalidCsv.jsx';
 
 export const getDataSample = async (filePath) => {
-  const parser = fs.createReadStream(filePath).pipe(parse({ columns: true, skipEmptyLines: true }));
+  const parser = parse({ columns: true, skipEmptyLines: true });
 
-  //* read the first line to get the file structure
-  for await (const record of parser) {
-    return record;
+  try {
+    const parsed = fs.createReadStream(filePath).pipe(parser);
+
+    //* read the first line to get the file structure
+    for await (const record of parsed) {
+      return record;
+    }
+  } catch (parseError) {
+    throw new Error(`${CSV_PARSE_ERROR}: {${parseError.code}} {${parseError.message}}`);
   }
 };
 
 export const getRecordCount = (filePath) => {
   return new Promise((resolve, reject) => {
-    const rs = fs.createReadStream(filePath);
-    const parser = parse({ columns: true }, function (err, data) {
-      if (err) {
-        reject(err);
-
-        return;
-      }
+    const parser = parse({ columns: true }, function (parseError, data) {
+      reject(`${CSV_PARSE_ERROR}: {${parseError.code}} {${parseError.message}}`);
```

Comment:
**issue**: is this function correct? It looks like it will always call reject and never get to resolve.